quilt/tools/build.py

```diff
@@ -106,3 +120,94 @@ def build_package(username, package, yaml_path):
         _build_table(build_dir, store, '', tables)
         if readme is not None:
             _build_file(build_dir, store, 'README', rel_path=readme)
+
+def generate_build_file(startpath, outfilename='build.yml'):
+    startbase = os.path.basename(startpath)
+    buildfiles = {startbase : {}}
+    buildtables = {startbase : {}}
+
+    def add_to_buildfiles(path, files):
+        try:
+            safepath = [_pythonize_name(dir) if dir != '.' else '.' for dir in path]
```

Comment:
Don't use built-ins like `dir`, `file`, etc. as variables. Even if it's a list comprehension, it'll break the built-in for the rest of the scope (in Python2, at least).